crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -164,6 +164,7 @@ pub struct Scope {
     pub(super) parent: Option<FileScopeId>,
     pub(super) kind: ScopeKind,
     pub(super) descendents: Range<FileScopeId>,
+    pub(super) contained_inside_try_block: bool,
```

Comment:
I think it would be slightly better if this were kept as temporary state in the scope stack in the index builder, rather than in the scope itself. I think it is only needed transiently in the builder, so we don't need to pay the memory cost of storing it permanently in the symbol table for every scope.

Reply:
I suspected I might get this feedback 😄 I decided to do this anyway as it seems much cleaner than the alternative. But I guess perf wins out!

Reply:
I pushed https://github.com/astral-sh/ruff/pull/13338/commits/c25592d095833eb1c9f1d901e08439ee9b5d6e2e